client/mm/wrapped_cex.go

```diff
@@ -0,0 +1,304 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package mm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"decred.org/dcrdex/client/asset"
+	"decred.org/dcrdex/client/mm/libxc"
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/calc"
+)
+
+type cexTrade struct {
+	qty         uint64
+	rate        uint64
+	sell        bool
+	fromAsset   uint32
+	toAsset     uint32
+	baseFilled  uint64
+	quoteFilled uint64
+}
+
+type cex interface {
+	Balance(assetID uint32) (*libxc.ExchangeBalance, error)
+	CancelTrade(ctx context.Context, baseID, quoteID uint32, tradeID string) error
+	SubscribeMarket(ctx context.Context, baseID, quoteID uint32) error
+	SubscribeTradeUpdates() (updates <-chan *libxc.TradeUpdate, unsubscribe func())
+	Trade(ctx context.Context, baseID, quoteID uint32, sell bool, rate, qty uint64) (string, error)
+	VWAP(baseID, quoteID uint32, sell bool, qty uint64) (vwap, extrema uint64, filled bool, err error)
+	Deposit(ctx context.Context, assetID uint32, amount uint64, onConfirm func()) error
+	Withdraw(ctx context.Context, assetID uint32, amount uint64, onConfirm func()) error
+}
+
+// wrappedCEX implements the CEX interface. A separate instance should be
```

Comment:
Please provide some basic documentation on this new interface, which apprears largely the same as `libxc.CEX`. Just a few sentences on the architecture.